openlibrary/plugins/openlibrary/js/carousels_partials.js

```diff
@@ -29,5 +29,13 @@ export function initCarouselsPartials() {
                 }
             }
         });
-    });
+    };
+
+    $('.loadingIndicator').removeClass('hidden');
+
+    if (document.readyState === 'complete') {
+        fetchRelatedWorks();
+    } else {
+        $(window).on('load', fetchRelatedWorks);
```

Comment:
I'm not sure about this code - it's already inside a jQuery ready event in https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/openlibrary/js/index.js#L114 

I think the issue here is that we are registering a window.load event /way after/ the load event has occurred.

My first thought it that calling `fetchRelatedWorks();` outside a window.load event should have exactly the same effect.

Reply:
> I think the issue here is that we are registering a window.load event /way after/ the load event has occurred.

I think that you are right about this.  Do you think that it would be better to remove the conditional and simply call `fetchRelatedWorks()`?

Reply:
I think so.
If that doesn't work it would indicate a more serious problem with how we load our code that I'd like to look into.